certification/pyxis/layers.go

```diff
@@ -0,0 +1,71 @@
+package pyxis
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	cranev1 "github.com/google/go-containerregistry/pkg/v1"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func (p *pyxisEngine) CheckRedHatLayers(layerHashes []cranev1.Hash) (*CertImage, error) {
+	ctx := context.Background()
+
+	layerIds := make([]string, 0, len(layerHashes))
+	for _, layer := range layerHashes {
+		layerIds = append(layerIds, layer.Hex)
+	}
+
+	log.Tracef("the layerIds passed to pyxis are %s", layerIds)
+
+	req, err := p.newRequestWithApiToken(
+		ctx,
+		http.MethodGet,
+		getPyxisUrl(fmt.Sprintf("images?page_size=500&page=0&filter=repositories.registry=in=(registry.redhat.io,registry.access.redhat.com) and uncompressed_top_layer_id=in=(%s)", strings.Join(layerIds, ","))),
+		nil,
+	)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	resp, err := p.Client.Do(req)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	if resp.Body != nil {
```

Comment:
This `if` statement is not needed, go guarantees a non-nil body.

https://pkg.go.dev/net/http#Response.Body

Reply:
removed

Reply:
@skattoju You still need the `defer resp.Body.Close()`, the `if` is what is not needed. Sorry this wasn't clear.

Reply:
re-added 😅